src/piton/transforms/stanford.py

```diff
@@ -16,6 +16,103 @@ def _move_date_to_end(
         return d
 
 
+def move_visit_start_to_day_start(patient: Patient) -> Patient:
+    """Assign visit start times of 12:00 AM to the start of the day (12:01 AM)
+
+    This avoids visits being pushed to the end of the day by e.g., functions that map
+    all events with midnight start times to the end of the day, such as `move_to_day_end`
+    """
+    for event in patient.events:
+        if (
+            event.start.hour == 0
+            and event.start.minute == 0
+            and event.start.second == 0
+            and event.omop_table == "visit"
+        ):
+            event.start = event.start + datetime.timedelta(minutes=1)
+
+            if event.end is not None:
+                event.end = max(event.start, event.end)
+
+    patient.resort()
+
+    return patient
+
+
+def move_visit_start_to_first_event_start(patient: Patient) -> Patient:
+    """Assign visit start times to be just before the first event
+
+    This function assigns the start time associated with each visit to be
+    either (1) one second before the start time of the first event on the
+    day of the associated with the visit, or (2) 11:58:59 PM if the visit
+    has no associated events or all events associated with the visit have
+    a start time of '00:00:00'. Events that occur on days prior to the
+    visit do not affect the visit start time.
+
+    NOTE: This function assumes that all non-visit events with the same
+    start time as the visit event (e.g., events with a start time at midnight
+    such as billing codes, in the case where visit events also have a midnight
+    start time) are moved to day end, otherwise there may be non-visit events
+    with start times before the visit start time.
+
+    The reason for assigning a start time of 11:58:59 PM is that it makes
```

Comment:
We operate in minute resolution by default, so this should be 11:58 PM

Reply:
Sounds good - I was back and forth about seconds vs. minutes but I think that's the right call for consistency.